lib/codegen/Decision.cpp

```diff
@@ -580,20 +361,20 @@ void makeEvalOrAnywhereFunction(KORESymbol *function, KOREDefinition *definition
   initChoiceBuffer(dt, module, block, stuck, fail, &choiceBuffer, &choiceDepth, &jump);
 
   int i = 0;
+  Decision codegen(definition, block, fail, jump, choiceBuffer, choiceDepth, module, returnSort);
   for (auto val = matchFunc->arg_begin(); val != matchFunc->arg_end(); ++val, ++i) {
     val->setName("_" + std::to_string(i+1));
-    subst[std::make_pair(val->getName(), val->getType())] = val;
+    codegen.store(std::make_pair(val->getName(), val->getType()), val);
     std::ostringstream Out;
     function->getArguments()[i]->print(Out);
     initDebugParam(matchFunc, i, val->getName(), cats[i], Out.str());
   }
-  addStuck(stuck, module, function, subst, definition);
+  addStuck(stuck, module, function, codegen, definition);
 
-  Decision codegen(definition, block, fail, jump, choiceBuffer, choiceDepth, module, returnSort);
-  codegen(dt, subst);
+  codegen(dt);
 }
 
-void abortWhenStuck(llvm::BasicBlock *CurrentBlock, llvm::Module *Module, KORESymbol *symbol, std::map<std::pair<std::string, llvm::Type *>, llvm::Value *> &subst, KOREDefinition *d) {
+void abortWhenStuck(llvm::BasicBlock *CurrentBlock, llvm::Module *Module, KORESymbol *symbol, Decision &codegen, KOREDefinition *d) {
```

Comment:
There may not be a special reason, but I'm curious: Why is codegen sent by reference, while the others are pointers?

Reply:
just convention. A bunch of llvm functions take and return pointers to llvm ASTs, so that's why the llvm namespace objects are pointers. We use smart pointers to keep track of the KORE ASTs, so that's why the KORE AST nodes are pointers. I just used a reference for Decision because it seemed simpler, given that we didn't have any compelling reason to use a pointer in that case.